src/Worker/Grpc/GrpcDurableTaskWorker.Processor.cs

```diff
@@ -237,6 +241,14 @@ async Task ProcessWorkItemsAsync(AsyncServerStreamingCall<P.WorkItem> stream, Ca
                         this.Logger.UnexpectedWorkItemType(workItem.RequestCase.ToString());
                     }
                 }
+
+                if (tokenSource.IsCancellationRequested || tokenSource.Token.IsCancellationRequested)
```

Comment:
Maybe I am missing something, but it seems unlikely this line would ever be `true`. If `IsCancellationRequested`, then more likely than not `stream.ResponseStream.ReadAllAsync` throw `OpeationCancelledException`.

Reply:
See above. We thought this behavior was an odd choice for the stream reader as well, but it's documented that it doesn't throw.